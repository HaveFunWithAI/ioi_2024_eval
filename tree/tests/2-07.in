2000
0 0 2 3 4 4 6 6 8 9 9 11 12 13 12 15 15 10 11 19 20 13 18 10 24 25 25 8 28 29 29 31 31 33 28 35 36 35 36 39 38 41 37 30 44 23 41 23 37 24 50 50 33 44 19 55 55 38 58 3 60 61 62 63 64 65 66 64 68 69 69 62 72 73 74 75 74 75 77 73 80 81 81 61 84 85 86 87 88 88 68 91 91 80 63 85 96 96 98 99 98 101 97 103 103 101 72 107 107 108 110 110 109 113 113 108 116 86 118 118 119 120 120 119 60 125 126 127 128 129 129 131 130 131 130 125 136 136 137 139 140 138 142 143 144 143 146 144 146 138 150 151 152 152 151 155 140 150 158 159 159 128 162 162 164 163 164 137 168 169 170 169 172 172 170 32 127 177 178 179 179 177 182 183 178 185 185 182 188 183 5 191 191 193 194 195 195 192 198 199 200 192 202 197 193 205 206 207 206 209 209 196 194 213 214 214 213 157 32 7 220 221 222 222 224 224 223 223 65 229 229 217 163 1 234 234 236 235 238 239 240 241 242 243 239 245 246 246 248 248 249 242 245 253 254 255 255 249 243 247 260 237 262 263 264 264 266 267 266 269 267 263 265 272 274 275 275 265 278 278 252 235 282 283 284 285 285 287 288 288 287 291 286 293 250 274 296 296 250 262 300 301 302 303 302 305 300 307 308 309 309 307 308 313 301 315 236 317 318 319 318 321 322 323 324 324 321 327 328 329 327 329 328 333 333 331 336 331 338 320 340 341 341 340 344 323 346 336 338 252 237 351 351 353 353 355 356 357 352 359 360 361 361 359 364 365 354 354 368 369 369 367 372 372 367 375 355 377 378 378 377 381 360 383 368 344 312 387 352 389 390 389 392 392 393 390 396 391 394 393 396 391 313 316 305 319 406 407 408 408 260 407 412 412 317 415 416 416 418 419 420 418 422 423 417 422 426 426 415 429 429 431 432 432 434 433 431 437 438 438 437 441 441 434 433 417 446 447 446 449 449 430 452 452 453 454 420 423 425 459 322 461 462 462 461 465 465 357 419 469 469 240 472 473 473 475 474 472 478 479 479 199 482 482 385 454 95 487 488 487 490 198 492 493 493 492 496 425 498 498 383 283 502 503 504 504 506 505 505 506 502 511 512 512 514 513 513 514 511 519 520 520 503 523 524 524 475 116 168 529 530 530 529 533 533 523 536 536 157 241 540 541 540 543 541 543 447 126 548 549 550 549 550 553 552 555 553 548 558 558 560 561 559 560 564 564 563 559 568 568 551 555 205 573 574 574 573 577 577 254 580 580 87 583 99 364 586 586 202 589 590 590 66 77 273 1 596 596 598 599 600 601 602 603 604 602 597 607 607 609 610 610 612 611 614 615 615 609 618 619 614 621 621 613 624 608 626 626 628 629 630 628 632 633 633 606 636 636 629 639 601 641 642 639 632 645 645 619 648 649 649 648 652 652 643 624 603 657 598 659 659 661 662 663 663 665 666 666 664 665 670 660 672 673 674 675 674 677 660 673 680 680 682 682 679 685 686 687 686 689 685 689 669 664 694 694 670 672 698 698 699 701 701 699 704 700 706 620 708 708 618 711 712 713 713 712 716 716 711 613 720 720 643 677 611 725 726 725 728 726 606 731 731 687 604 681 669 719 738 738 662 741 742 743 741 743 742 747 747 745 750 704 728 745 754 754 642 757 757 691 760 760 675 608 764 765 766 767 767 766 770 765 772 773 773 770 599 777 778 779 780 781 781 780 778 785 786 787 779 789 789 791 791 790 785 784 795 797 790 797 784 795 802 802 777 805 805 806 808 809 809 807 812 813 813 808 816 816 657 691 820 820 787 641 824 824 825 825 826 786 830 830 155 700 834 84 836 837 838 839 839 836 842 842 844 843 846 843 848 845 848 845 846 844 854 7 856 857 857 859 860 858 856 863 864 865 865 858 868 868 864 871 871 862 859 875 291 474 764 879 880 881 881 879 884 885 884 887 885 887 880 891 891 706 303 490 875 217 430 899 900 900 899 903 903 253 906 907 906 909 142 911 912 912 911 915 915 583 907 620 920 920 97 923 496 221 926 927 927 926 930 930 381 459 203 935 935 385 837 939 940 939 942 940 942 95 946 947 946 949 949 238 952 952 954 955 953 957 958 959 957 958 962 962 954 965 966 967 955 961 959 953 972 973 974 961 976 976 974 956 980 980 969 983 983 967 970 965 988 989 989 969 992 992 966 995 995 18 998 998 860 30 1002 1002 356 1005 1005 597 1008 1009 1010 1010 1011 1013 1013 1012 1016 1009 1018 1019 1019 1018 1022 1023 1024 1023 1026 1026 1021 1029 1029 1016 1022 1033 1034 1034 1033 1037 1024 1011 1040 1040 1037 1008 1044 1045 1046 1047 1047 1046 1050 1050 1045 1053 1054 1053 1056 1054 1021 1059 1059 1020 1062 1062 346 551 247 1067 1067 863 1070 1071 1070 1073 1073 1071 2 1077 1078 1079 1080 1081 1082 1083 1084 1084 1081 1087 1088 1089 1088 1091 1091 1089 1080 1095 1096 1097 1098 1083 1100 1078 1102 1103 1104 1105 1106 1106 1108 1107 1102 1104 1112 1113 1114 1113 1116 1116 1107 1111 1120 1120 1121 1122 1124 1125 1125 1122 1121 1123 1130 1123 1132 1108 1129 1135 1129 1137 1137 1132 1105 1141 1142 1142 1114 1096 1146 1147 1147 1111 1150 1151 1152 1153 1153 1151 1156 1157 1152 1159 1159 1157 1150 1163 1164 1165 1165 1163 1168 1169 1164 1171 1168 1169 1171 1173 1112 1177 1178 1177 1180 1180 1128 1183 1183 1141 1186 1186 1135 1156 1190 1190 1079 1193 1194 1193 1196 1197 1198 1199 1198 1197 1202 1203 1195 1205 1206 1205 1208 1202 1210 1208 1194 1213 1214 1215 1215 1214 1218 1218 1201 1206 1203 1196 1224 1225 1225 1227 1227 1226 1224 1231 1232 1232 1226 1201 1146 1237 1237 1130 1199 1082 1242 1243 1242 1245 1245 1243 284 1249 1250 1251 1251 1250 1254 1254 1249 1257 1258 1258 1257 1261 1261 862 286 1265 320 1267 1268 1268 1267 1271 1271 269 679 1275 1276 1277 1277 1275 1280 1281 1280 1283 1281 1283 1276 1287 719 1289 1289 141 1020 1293 1293 806 1296 1297 1297 1296 1300 1300 826 315 1304 1304 109 1307 1307 627 1310 1311 1311 1310 1314 1314 139 1317 1318 1318 1317 1321 1321 196 5 1325 1326 1327 1328 1328 1330 1330 1326 1333 1333 1335 1336 1336 1334 1339 1339 1329 1335 1343 1325 1345 1346 1347 1347 1349 1349 1348 1346 1353 1353 1355 1348 1355 1354 1345 1360 1361 1362 1360 1364 1362 1361 1367 1367 1327 1370 1371 1371 1370 1374 1374 1364 1377 1354 1329 1365 1365 1334 1383 1383 272 1386 1387 1386 1389 1389 1387 627 1393 1394 1394 1393 1397 1397 1173 750 203 1402 1402 94 854 834 1231 1408 1408 600 1411 1412 1413 1414 1412 1414 1416 1418 1418 1413 1421 1416 1423 1423 1421 1265 293 207 1087 1430 1431 1430 1433 972 1435 1436 1436 1435 1439 1439 681 956 1443 1443 1097 1446 1446 1103 1449 1450 1451 1452 1450 1454 1452 1454 1457 1457 1451 1455 1455 1460 1460 1449 1465 1466 1467 1467 1465 1470 1471 1471 1470 1474 1466 1476 1476 1474 812 1480 1480 1012 1483 1484 1483 807 1487 1488 1488 1487 1491 1491 39 1100 1343 630 312 1498 1498 1210 197 923 141 988 1505 1505 58 1077 1509 1510 1509 1512 1512 1514 1515 1516 1516 1515 1518 1520 1519 1519 1523 1524 1517 1526 1526 1510 1529 1530 1531 1532 1533 1533 1532 1536 1529 1536 1538 1540 1540 1541 1543 1543 1542 1546 1542 1548 1548 1523 1551 1514 1553 1554 1555 1556 1555 1558 1554 1560 1558 1511 1563 1564 1565 1566 1566 1564 1569 1569 1571 1570 1570 1565 1575 1575 1571 1563 1579 1579 1580 1582 1581 1580 1585 1546 1584 1551 1581 1590 1590 1584 1522 1594 1560 1596 1561 1522 1599 1596 1582 1553 1603 1604 1605 1603 1607 1608 1604 1610 1610 1605 1607 1614 1614 1585 1561 1513 1619 1620 1621 1621 1623 1620 1625 1626 1622 1625 1629 1629 1623 1622 1524 1530 1635 1636 1637 1636 1639 1635 1641 1642 1642 1641 1645 1639 1518 1648 1626 1513 1651 1652 1653 1654 1652 1656 1657 1657 1651 1660 1661 1662 1662 1661 1665 1665 1660 1668 1669 1668 1671 1669 1556 1648 1645 1531 1677 1678 1677 1680 1680 1678 1637 1608 1511 1686 1687 1687 1689 1690 1689 1692 1692 1688 1695 1695 1690 1688 1699 1699 1654 1671 1538 1704 1704 1705 1707 1705 1709 1706 1711 1711 1707 1706 1715 1715 1709 1517 1719 1719 1594 1599 158 1724 1724 387 909 772 1729 1729 1520 970 612 1734 1735 1735 1734 1738 1738 453 375 1656 1743 1743 1095 1746 1746 1748 1748 1750 1750 1749 1749 1747 1755 1755 1747 1758 1758 488 1044 1762 1763 1762 1765 1766 1764 1765 1769 1769 1764 1766 1763 1774 1774 973 1777 1777 200 589 1781 1781 94 1287 563 1124 1787 1787 406 1790 1791 1791 1790 1794 1794 273 478 1798 1798 1686 1801 1802 1803 1803 1801 1806 1802 1808 1807 1806 1811 1811 1807 1808 552 1816 1816 394 1431 1411 1821 1822 1823 1821 1825 1826 1822 1823 1826 1825 1831 1828 1831 1828 1619 1836 1836 1837 1838 1840 1840 1838 1843 1843 1837 1846 1846 1839 1839 519 1851 1851 1178 661 1855 1856 1857 1858 1857 1860 1858 1856 1863 1864 1855 1866 1867 1868 1868 1864 1863 1872 1872 1867 1875 1860 1866 1878 1879 1879 1878 1882 1875 1882 1653 1886 1886 838 1889 1889 1195 1892 1893 1893 1892 1896 1896 1433 188 220 1901 1902 1903 1903 1901 1906 1907 1907 1056 20 1377 282 1913 1913 1915 1916 1917 1918 1918 1917 1921 1921 1916 1924 1925 1925 1924 1928 1928 1914 1931 1932 1932 1934 1934 1933 1931 1938 1939 1939 1915 1942 1943 1944 1944 1943 1947 1947 1942 1950 1951 1951 1950 1954 1954 1914 1957 1958 1958 1959 1960 1960 1959 1933 1938 1966 1966 1957 1969 1970 1969 1972 1972 1970 316 1213 1977 1978 1978 1977 1981 1981 1128 1984 1984 1906 1987 1987 365 1098 947 1902 1993 1993 1541 1996 1996 561
798942 280504 338913 271126 476442 184178 572093 977814 895573 9378 328107 301817 896736 702886 309353 278339 151301 677927 283170 764074 614540 544793 315339 952815 256044 386433 854130 746464 782957 983531 118420 273534 664594 474350 130834 631620 669537 123669 704795 242934 836280 660329 811733 105344 24135 65210 108060 810272 112258 630672 659359 584376 219498 245182 446596 17822 380159 245048 544175 203872 567391 177244 404129 14529 573606 729360 195403 352351 463120 838533 584057 237112 746901 15970 991276 107842 116861 581931 766766 556849 456233 416145 857133 514097 134940 168457 771140 178537 636316 274286 35010 308260 255919 278051 76553 835593 590177 433136 872737 980952 664990 121617 576811 180358 57858 270394 771420 831750 22021 493316 541552 232255 970157 708624 238846 29420 413518 385260 156190 116780 420077 599468 737837 481164 959432 45925 924285 31889 456335 554887 822106 925026 580355 919536 659421 438152 86010 857241 350404 971438 543831 493673 187217 191817 638575 940280 62374 626317 987498 252087 827743 857885 325590 970519 495909 902063 52172 688844 247603 982508 303459 327798 211543 65121 404385 205710 826570 316192 318742 956976 548697 558291 97558 299594 446625 529743 130169 561183 710647 419035 716019 112148 416099 379774 620838 402593 393113 884247 20714 708833 3059 362200 611631 495518 310741 845784 848348 677669 368886 140858 616295 140668 401251 82480 577559 257300 379506 262186 965593 341780 576373 620710 526865 8812 850946 881280 745310 929014 267510 408023 73399 773583 210078 665339 559489 657024 115289 447299 819487 632421 765735 690379 847681 379204 55870 888289 741825 132588 71697 233877 205164 793689 24694 342650 10946 930532 794352 522697 296388 582989 338121 677125 789626 541404 46602 739283 140489 586842 533005 829870 348247 480364 421547 6005 961355 569355 486166 728410 721622 858693 637014 576256 410716 788189 272985 633619 298645 675286 530296 564947 910223 450576 258452 466366 371957 846611 132374 278504 42205 176079 674983 904369 526775 725893 182160 473406 954017 205777 207572 519093 760347 922554 120110 821774 602556 664978 228539 958135 524341 596844 694774 354963 982569 183131 993981 509122 633903 354346 478690 158176 628094 552322 745874 869571 49213 370950 136336 187594 770220 79605 94298 570872 822921 699359 797951 843395 458498 336768 238386 30881 112107 945264 758565 676021 170199 679910 531173 819197 534692 10860 56941 250636 467376 604978 467938 921986 211452 601162 793003 519554 385051 220966 293897 342329 290062 380647 263967 235411 752064 536188 921394 135642 671232 513821 508177 436141 81471 536147 146339 378461 298806 212354 768983 146828 482476 292789 32042 142410 743524 124224 382449 542521 900520 968511 421648 360471 950107 78491 834530 53190 502068 368346 452772 678580 231088 179570 831896 930489 748171 569657 765509 105522 605043 518560 241565 206813 262770 331891 963188 18452 163866 832323 522447 75545 877861 857152 243236 454389 264326 624874 440528 680852 214466 205845 513478 61686 665728 566618 643103 657144 268151 109300 594414 672336 871378 370185 213909 811994 870270 359455 907572 657753 813079 522154 439982 701575 966653 519360 472453 338326 763707 253727 962475 193789 951387 504293 604923 9606 13162 620668 882126 390639 806908 941663 540984 832327 182497 807264 470078 451807 755761 111220 391824 733992 987825 132204 230876 152852 242180 508046 548013 867004 289410 442584 793955 900257 524374 499192 201066 318042 817911 969186 587810 198177 982255 306953 150137 505046 30165 198206 379446 519702 234438 120680 668517 957970 698645 456389 954746 858048 504216 260183 941820 260862 369628 47259 971246 406102 517609 370663 972309 139236 206002 652773 701151 544262 359137 630681 192802 136 491687 432346 795589 853509 823912 714998 717748 550456 418398 526317 687186 407292 594329 271185 803005 453374 271758 108881 273563 423750 154295 53200 1591 554745 592641 186612 15896 24084 65863 527076 541341 958992 671946 55750 200706 555885 313756 988453 191206 106841 13530 610800 848086 145585 736091 506251 935852 900058 987544 435232 916705 483509 576020 224855 148125 905178 69006 872963 428444 600412 848151 685169 870880 449706 564141 638904 838279 395728 817578 715296 907995 451291 866965 825989 861955 561937 869040 344089 948896 456945 387545 965818 370800 147683 735844 544146 523517 931685 632999 451265 106408 774266 569805 180865 528504 996428 171471 590032 170381 61781 302518 997919 181611 913950 300552 997331 165599 598461 925255 74277 221734 834539 202656 339649 455838 412863 304785 368730 460285 412322 572904 963534 247598 973356 883321 978602 655275 577081 157904 515204 458046 286056 621450 344559 652596 852063 43273 51693 250718 882822 260404 875532 461838 371741 369849 962771 954749 763407 141019 743516 799945 46979 651340 190201 258192 481089 702238 24848 46408 559971 671175 838261 756496 117906 750046 349243 639311 176604 924659 204338 745042 32378 965865 509917 952638 549041 124748 443461 224700 302729 387879 115392 396795 304098 724062 694757 638116 653934 180161 155415 141460 729455 13595 81694 971667 635193 959080 584617 322672 470154 475868 961833 829126 935918 424766 307869 674136 972697 861848 264163 313683 891275 512965 93420 507848 499730 364432 475900 960433 723009 102878 376445 165028 65417 747997 738487 811298 420831 719274 76448 142217 407377 537323 142348 692966 25255 709758 653602 155212 908359 82779 588965 478681 327101 907325 449669 653877 683042 32135 224618 384305 994406 621565 71920 411331 369180 51936 473234 324252 524654 383204 949873 418281 575632 397192 295247 737175 216202 602956 316939 944597 815860 723096 238499 440469 275079 314313 313773 929862 144116 967048 365941 826104 869878 293748 889344 918179 780321 695409 942147 135916 737039 750994 781034 993377 475292 664582 760094 176731 651244 38434 421262 601080 408059 569613 15406 134692 863553 138011 718474 516631 13562 201358 999890 429029 917427 309898 763130 60001 5500 162409 139316 953918 959675 301151 845618 365878 464138 184618 719377 460482 190774 450851 137241 401717 944016 38868 630502 53928 59501 608073 682848 519696 636841 422450 748859 712147 663263 256151 235887 892205 766519 180036 637881 796823 580223 518740 730804 984545 373242 665304 917747 651391 904223 154447 192517 548176 203491 534089 442777 639178 338027 152417 874251 845067 359947 645238 800604 693350 322519 841569 170970 317953 580359 40776 511976 48671 541023 597785 948354 527187 172885 861898 936291 72128 541774 77925 911806 923423 933825 278043 918454 262499 642612 988656 281496 666098 145563 635570 107843 543246 909424 347759 311866 910190 526063 374749 817637 29725 193199 600395 222186 395325 654973 401461 689290 347861 622779 660073 419587 521669 733389 508546 828286 293403 293981 676078 594832 68209 190545 676441 803252 640398 116242 18071 239352 777901 829849 741178 792511 223907 96196 998416 381390 938552 673289 864624 381667 961292 492251 768223 958029 876396 936466 800352 942923 684908 24403 456190 139866 809716 721002 851906 381760 469708 385058 715737 321178 909292 796962 147415 272960 412655 207446 453631 473438 712304 122757 399191 691459 128675 255566 344678 190072 91719 332532 58015 823707 270450 387374 538990 217268 154028 625746 148714 353705 739396 275778 163585 885202 569893 630597 586390 694336 882887 675528 29258 650442 292079 437996 265459 300235 739262 963005 83377 24132 935230 521016 879346 250838 391266 570214 334197 515401 245291 673895 327798 440162 605830 17619 449407 424891 167184 855070 545760 305735 265421 474564 896816 861718 298727 747921 561149 872058 888093 525167 646023 328412 270960 11420 478659 702750 51523 924451 351329 819058 898357 919989 960295 893184 142752 510846 661803 176925 26919 623116 818129 224064 660235 481764 633112 695832 683810 274107 337806 501479 31994 274084 874394 411874 95175 58143 474581 949031 996099 59831 511252 603164 986670 647898 614990 58731 234907 73950 57950 239171 595952 179209 793080 927856 960599 790046 46602 139184 404153 406877 943903 318105 241042 645134 610928 788315 121344 499982 23853 296169 960735 14210 877523 559585 249492 398047 723576 292959 529613 972081 122177 219253 222204 439214 412307 481584 785441 651811 437437 252869 914006 619265 897508 819663 38279 302555 43331 460886 401244 403617 952341 68477 363189 752475 154118 342234 176584 565333 212107 310662 470183 337304 3225 537139 205752 439267 68471 218268 544039 992920 288790 213057 847651 387235 68746 163792 990388 292991 139993 312349 103864 524476 355930 179495 354383 62413 416015 81813 17748 788472 731447 883111 830803 900153 942921 741206 906897 164372 661103 948690 926941 972581 468329 79589 768021 454191 130078 867266 326546 334963 488080 487476 922054 221795 62074 514431 894033 256428 616111 429094 818713 86741 440202 398767 821520 500891 182280 376018 956490 418283 163843 972967 458512 662684 871123 211631 287064 188351 292753 12706 39083 194735 611158 264825 629826 783834 550625 217110 683597 338848 249882 710047 589182 730394 811478 964275 552679 888680 794901 387954 834332 868791 786040 966980 787373 191615 229046 114187 459106 270202 117491 549704 316276 930974 347985 470367 812643 287836 74861 307427 120115 131622 607522 678668 195104 683092 587459 93731 982827 669875 554924 753449 98744 305490 310142 748105 889121 892405 956866 977019 813416 392602 265562 522663 817869 158031 204373 579550 293207 319516 821317 966046 73181 630188 927892 848977 847748 818021 839362 880237 375825 587743 886602 459644 86675 788861 65239 426585 262325 464979 224913 678855 30802 821597 421456 318190 883596 661301 694931 980437 250820 894322 755205 483156 467464 879646 26734 532097 141021 670980 602736 313258 818976 223102 632519 799724 24281 623727 255369 366771 354726 694013 185677 149080 174809 908019 366160 848419 820547 598243 848901 123914 869628 649184 116621 649237 882125 923440 34264 21869 397636 744290 560420 731840 880540 911693 335590 946565 475932 245094 293948 466136 543301 44532 686959 439404 968259 935902 580916 927629 494 775315 510372 736537 827507 822230 24102 289283 364261 653356 815346 948458 794444 849083 291797 522125 939214 506565 514659 409734 846409 305940 884610 325971 551765 514702 672300 938042 793187 260558 215811 864093 344014 617313 890106 768815 204856 309742 93524 497301 193719 581895 659551 528799 527465 989390 888371 334584 982333 380139 986767 216530 543836 251635 830908 123337 498390 543545 131624 905321 473324 287958 315327 487794 145749 578622 838447 80267 421368 467355 501007 179724 524800 772898 560717 933852 927711 716519 713490 3209 58217 798772 633656 352040 916479 736420 278229 745417 539124 577811 590513 919306 638472 586093 79625 404114 81177 938416 934934 572569 710173 344236 436359 46553 472996 87584 422185 823632 44081 701327 277189 452967 617624 927703 792925 362194 232425 824455 409791 997541 841140 805125 575671 269991 228728 223392 248079 591040 745079 892430 151461 510202 365307 364709 937241 900518 516472 830857 904882 1859 829749 84074 340332 613475 790292 857491 318742 346565 334254 729296 681584 163571 789590 395619 941277 161789 335658 745804 72179 465532 159436 460533 494708 290981 665574 738535 146220 362974 990907 512267 251386 645794 509763 128583 442005 706870 690436 714776 891442 288048 550594 991579 642037 703778 745288 826317 493189 360864 873500 501372 309392 368361 340121 201534 123775 804645 163399 9058 220166 112054 944098 505161 589332 508112 373898 586205 726298 953102 174599 81173 917075 592524 995491 456007 943982 361275 654672 373662 212219 657411 377510 872991 676503 661883 854311 215984 320597 461804 189141 852633 782548 126752 1252 227018 777332 217342 539441 704567 142108 955112 730972 452439 425159 850621 668931 212685 580118 855207 956488 743336 17921 265203 883168 402632 123092 13452 116010 259281 715281 437153 116232 619804 224119 627340 386824 255427 182821 742648 496382 60495 201643 755718 954293 168928 60357 380672 718250 197447 519114 189050 802989 643629 148097 572180 606015 681795 713336 426189 237472 824202 893438 415349 690420 335412 850677 577189 782113 376814 897362 499462 869328 601851 267816 986816 12327 376546 209482 539098 81071 451610 848257 115717 579901 844652 214903 861161 353766 950004 868841 221493 379601 589297 61016 174321 266870 880154 692388 450009 706904 156802 119852 583602 261162 405038 468336 440952 198302 582166 578860 419701 442657 586152 317745 240071 167281 328449 85149 653611 698700 626259 372585 205046 707606 148986 454112 402709 142969 87060 280509 416418 222547 861324 652281 998596 20449 366670 786894 256928 70365 24753 152240 839451 797507 257698 228630 570852 227898 720399 248278 280820 253960 405950 922377 28226 744306 111977 61487 214010 585471 150504 851481 810947 31646 565550 656129 551562 326000 759953 679763 713077 335685 106166 581812 639680 553555 513039 825041 980037 46856 807177 825781 659368 1016 506626 483535 352422 945054 985124 123557 751406 36549 945983 613033 210773 643522 85970 929431 701262 96191 377330 951315 272167 619057 33818 903369 488206 751658 908057 462801 164880 455635 417161 389338 29447 455747 366722 888213 356106 857837 980871 945177 466463 116161 861795 360344 573873 364438 592431 45623 422591 904794 168010 489291 673022 272439 379624 477992 546008 815149 69168 537272 920230 533787 850463 992420 247865 660498 321369 609859 70766 604149 231347 87321 604882 237309 65739 682713 50031 32218 930594 974978 52289 58045 188820 137929 405824 713574 619730 239250 147094 345641 985441 89177 14964 773930 837939 137361 546398 160380 990407 395531 456493 992464 82648 797201 459666 155372 87893 260613 486277 856715 884576 560742 918416 464072 338615 517019 727613 538355 638325 62971 887640 928301 342912 805061 774214 110028 980232 227916 402718 960786 205568 756152 218557 148567 768829 320304 849289 197103 507159 947175 318454 175398 604917
10
239749 420297
209559 386641
387466 803070
448330 795767
5321 57828
35555 234530
3949 379662
23986 47727
58020 867459
436054 677486
