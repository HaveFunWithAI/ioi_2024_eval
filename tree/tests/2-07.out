209059603741141
179481179940217
320133949987194
389284374280293
3450473671431
24432829075090
2164055762728
19957133648605
36369151512141
394914390055062
